File: backend/src/main/java/net/zorphy/backend/site/jolly/dto/game/GameState.java
package net.zorphy.backend.site.jolly.dto.game;

import net.zorphy.backend.site.core.http.dto.PausableGameState;
import net.zorphy.backend.site.core.http.dto.PauseEntry;
import net.zorphy.backend.site.core.http.dto.SavableGameState;
import net.zorphy.backend.site.jolly.dto.RoundInfo;

import java.time.Instant;
import java.util.List;

public record GameState(
        boolean isSaved,
        List<PauseEntry> pauseEntries,
        Instant startTime,
        GameConfig gameConfig,
        List<RoundInfo> rounds
) implements SavableGameState, PausableGameState {

    @Override
    public GameState withSaved(boolean saved) {
        return new GameState(saved, pauseEntries, startTime, gameConfig, rounds);
    }
}

File: frontend/src/app/sites/jolly/dto/game/GameState.ts
import {GameStateBase} from "../../../core/http/dto/GameStateBase";
import {GameConfig} from "./GameConfig";
import {RoundInfo} from "../RoundInfo";
import {SavableGameState} from "../../../core/http/dto/SavableGameState";
import {PausableGameState} from "../../../core/http/dto/PausableGameState";

export interface GameState extends GameStateBase, SavableGameState, PausableGameState {
    gameConfig: GameConfig,
    rounds: RoundInfo[]
}